### DISCUSSION TITLE
Need Help with White Labeling a MAUI App from Existing Project

### DISCUSSION THREAD

**Discussion by satish-dnv** [dotnet/maui#22198] [Q&A] [answered]
Hi everyone,

I'm looking for guidance on how to white label a MAUI app from an existing project. Is there a specific process or packages available for this purpose? I have an existing app project and would like to customize it with different branding or themes for different clients. 
I found this article (https://xamgirl.com/white-labeling-in-xamarin-forms-applications/) which gives some ideas about how to white label a Xamarin app. However, I'm facing conflicts with the AndroidManifest and Info.plist files. How can I configure different manifest files according to the app? Any help or suggestions on how to achieve this would be greatly appreciated. 

Thank you!

**Comment by developer9969** [comment, marked as answer]
We white label our app and all you have to do

-  all the views and viewmodels in a separate Maui class libraries 
- create one app for each client that references the shared class library 
- the only thing that changes is your style 
- All the colors and other styles will be in the main project
- Use resource files for each client 
- 
I think above gives u an idea

**Reply by satish-dnv** [reply]
Thank you!

**Comment by kdbhalala** [comment]
I have more than 12 Clients that I have to create 12 apps in maui project right.
That is a bit over work I guess compared to Xamarin forms.
if you follow above tutorial it is a simple solution.
@developer9969

**Reply by developer9969** [reply]
@kdbhalala  that is one way , there are other ways to do that - you could have a single app and pipeline  changing things when building the app.. etc.. or when you do a build you can run a script that does things...

**Comment by saamerm** [comment]
@satish-dnv I don't know if you're still looking for a solution, here's some suggestions from personal experience:
- You can use different GitHub branches for each white label brand. That way you can just pull the changes into each brand branch from the main/develop branch, and it works pretty well.
- However, Shell Scripting is my personal favorite. I just [published an article](https://medium.com/@prototypemakers/quickest-way-to-white-labeling-maui-ios-android-apps-using-shell-scripts-0a1a6b951fb3) and [recorded a video](https://www.youtube.com/watch?v=UR2FNq4_qGw) that you can access for free on how to use shell scripting for white labeling with MAUI applications.

I have explained it in a lot of details in my article, but in essence, in your MAUI app’s folder/directory create a new file `brand1.sh` with this:
```
grep -rl 3071CF . | grep -v "brand1.sh" | xargs sed -i "" -e 's/3071CF/CE102C/g'
grep -rl BeAware . | grep -v "brand1.sh" | grep -v ".git" | xargs sed -i "" -e 's/BeAware/Brand1/g'
grep -rl com.tfp . | grep -v "Info.plist" | grep -v ".git" | grep -v "brand1.sh" | grep -v "MauiProgram" | grep -v "MainActivity" | grep -v "WebAuthenticatorActivity" | grep -v "AppDelegate" | grep -v "AccessTokenService" | grep -v "LoginPage" | xargs sed -i "" -e 's/com.tfp/com.brand1/g'
cp -a brand1/. BeAware
```
Create a new directory `brand1`, and put the images you want to replace in the right sub-directories/sub-folders using the same file name, format & dimensions. Then run the shell script using `sh brand.sh`

**Reply by satish-dnv** [reply]
@saamerm Thank you for suggestion. I'll have a look into that article.

**Comment by ewerspej** [comment]
I wrote an article on the topic as well called ["White labeling .NET MAUI apps"](https://blog.ewers-peters.de/white-labeling-net-maui-apps) which you may find useful. This works well for smaller projects and few clients. If you have many clients, you may want to use CI/CD pipelines to provide the required information, e.g. which client to build for, by passing in compile constants.
